src/Lecture1.hs

```diff
@@ -94,7 +110,9 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+subString :: Int -> Int -> [a] -> [a]
+-- Get all the items at those indexes, drop the ones that don't exist (@mapMaybe@ filters out the @Nothing@s and converts @Just a@s into @a@s)
+subString start end str = mapMaybe (str !?) [start..end]
```

Comment:
This looks correct. Unfortunately, this is a very slow implementation because index access in lists has linear complexity 😞